docs/Wrappers.md

```diff
@@ -0,0 +1,189 @@
+    ```javascript
+    import { WrapperService} from '@daostack/arc.js';
+    const wrapperTypes = WrapperService.wrappersByType;
+
+    for (var schemeWrapper of wrapperTypes.schemes) {
+      console.log(`${schemeWrapper.friendlyName} is at ${schemeWrapper.address}`);
+    }
+    ```
+
+!!! tip
+    `ContractWrappersByType.allWrappers` is an array of all of wrappers.
+
+## Get a wrapper at a given address
+
+You can use a wrapper's factory class to obtain a wrapper for a contract deployed to any given address:
+
+```javascript
+import { UpgradeSchemeFactory} from "@daostack/arc.js";
+const upgradeScheme = await UpgradeSchemeFactory.at(someAddress);
+```
+
+!!! info
+    `.at` will throw an exception if it can't find the contract at the given address.
+
+!!! tip
+    `ContractWrapperFactories` is an alias for [WrapperService.factories](api/classes/WrapperService/#factories)
+    ```javascript
+    import { ContractWrapperFactories } from "@daostack/arc.js";
+    const upgradeScheme = await ContractWrapperFactories.UpgradeScheme.at(someAddress);
+    }
+    ```
+
+Another way to get a wrapper at a given address is using [WrapperService.getContractWrapper](api/classes/WrapperService/#getContractWrapper).  This is most useful when you have both contract name
+and address and wish to most efficiently return the associated wrapper, or undefined when not found:
+
+```javascript
+import { WrapperService } from "@daostack/arc.js";
+// returns undefined when not found, unlike the factory `.at` which throws an exception 
+const upgradeScheme = await WrapperService.getContractWrapper("UpgradeScheme", someAddress);
+}
+```
+
+## Deploy a new contract
+
+You can use a wrapper's factory class to deploy a new instance of a contract and obtain a wrapper for it:
+
+```javascript
+import { UpgradeSchemeFactory} from "@daostack/arc.js";
+const newUpgradeScheme = await UpgradeSchemeFactory.new();
+```
+
+!!! tip
+    `ContractWrapperFactories` is an alias for [WrapperService.factories](api/classes/WrapperService/#factories)
+    ```javascript
+    import { ContractWrapperFactories } from "@daostack/arc.js";
+    const newUpgradeScheme = await ContractWrapperFactories.UpgradeScheme.new();
+    }
+    ```
+
+## Obtain a DAO scheme's parameters
+
+Although you can always register your own schemes with a DAO, whether they be totally custom non-Arc schemes, or redeployed Arc schemes, by default a DAO is created with Arc schemes that are universal in the sense that the code is implemented in one place, without redundancy.  But every scheme registered with a DAO is configured with a particular set of parameter values, and references such as proposals. All of these are stored in the DAO's own controller where each universal scheme is able to find them.  (If the controller is the Universal Controller then the parameters and data are keyed by the DAO's avatar address.)
```

Comment:
they are universal in the sense that its code can be usable by any dao which can be registered to it

Reply:
Done